BUG/TST: cluster: incorrect test for `seed` param of {`kmeans`, `kmeans2`}

### Describe your issue.

The test added in gh-13972 by @AtsushiSakai incorrectly compares `res1` to itself rather than `res2` (twice), so does not test the correctness of the `seed` parameter.

Once the test is corrected to compare `res1` and `res2`, it actually fails due to the elements of the array being in different orders (see the CI of gh-19682).

### Reproducing Code Example

```python
def test_kmeans_and_kmeans2_random_seed(self):

        seed_list = [1234, np.random.RandomState(1234)]

        # check that np.random.Generator can be used (numpy >= 1.17)
        if hasattr(np.random, 'default_rng'):
            seed_list.append(np.random.default_rng(1234))

        for seed in seed_list:
            # test for kmeans
            res1, _ = kmeans(TESTDATA_2D, 2, seed=seed)
            res2, _ = kmeans(TESTDATA_2D, 2, seed=seed)
            assert_allclose(res1, res1)  # should be same results

            # test for kmeans2
            for minit in ["random", "points", "++"]:
                res1, _ = kmeans2(TESTDATA_2D, 2, minit=minit, seed=seed)
                res2, _ = kmeans2(TESTDATA_2D, 2, minit=minit, seed=seed)
                assert_allclose(res1, res1)  # should be same results
```


### Error message (once corrected)

```shell
E   AssertionError: 
E   Not equal to tolerance rtol=1e-07, atol=0
E   
E   Mismatched elements: 4 / 4 (100%)
E   Max absolute difference: 2.5019998
E   Max relative difference: 3.09565287
E    x: array([[-3.063333,  3.31023 ],
E          [-1.947611,  0.80823 ]])
E    y: array([[-1.947611,  0.80823 ],
E          [-3.063333,  3.31023 ]])
```


### SciPy/NumPy/Python version and system information

```shell
`scipy` @ `main`
```

--- a/scipy/cluster/tests/test_vq.py
+++ b/scipy/cluster/tests/test_vq.py
@@ -1,5 +1,6 @@
 import warnings
 import sys
+from copy import deepcopy
 
 import numpy as np
 from numpy.testing import (
@@ -404,13 +405,15 @@ class TestKMean:
         ]
 
         for seed in seed_list:
+            seed1 = deepcopy(seed)
+            seed2 = deepcopy(seed)
             # test for kmeans
-            res1, _ = kmeans(TESTDATA_2D, 2, seed=seed)
-            res2, _ = kmeans(TESTDATA_2D, 2, seed=seed)
-            assert_allclose(res1, res1)  # should be same results
+            res1, _ = kmeans(TESTDATA_2D, 2, seed=seed1)
+            res2, _ = kmeans(TESTDATA_2D, 2, seed=seed2)
+            assert_allclose(res1, res2)  # should be same results
 
             # test for kmeans2
             for minit in ["random", "points", "++"]:
-                res1, _ = kmeans2(TESTDATA_2D, 2, minit=minit, seed=seed)
-                res2, _ = kmeans2(TESTDATA_2D, 2, minit=minit, seed=seed)
-                assert_allclose(res1, res1)  # should be same results
+                res1, _ = kmeans2(TESTDATA_2D, 2, minit=minit, seed=seed1)
+                res2, _ = kmeans2(TESTDATA_2D, 2, minit=minit, seed=seed2)
+                assert_allclose(res1, res2)  # should be same results

